### ISSUE TITLE
Refactor to use @angular/cdk overlays 

### ISSUE THREAD

**Issue by badre429** [ng-select/ng-select#978]
refactor ng-select to use **@angular/cdk** overlay

**Comment by varnastadeus** [comment]
can you elaborate why this should be done?

**Comment by maxkorz** [comment]
@varnastadeus using this package will allow you to use well-tested and proven code that is used by hundreds of developers around the world. It is likely that in the future many plugins will use this package, which will lead to a decrease in the final build size for all users. You won't have to write and maintain code for things like: Accessibility (focus trapping, navigation with keyboard), Overlay (auto positioning on scrolling), Virtual scrolling and Bidirectionality (respond to change in the application's LTR/RTL layout direction)

**Comment by varnastadeus** [comment]
ng-select is well tested too, and current implementation doesn't have any issues. As I can see, the only use case for cdk's overlay would be to close dropdown on outside click, which has major drawback since you need to click twice if you want to open another select on the same page (https://github.com/angular/material2/issues/9320). Another concern is being dependent on 3rd party library. Currently we have zero dependencies and we would like to keep that way.

We had "overlay" like implementation before and we sticked with current one since it is more user friendly. Currently I don't see any benefits why we should use cdk's overlay.

**Comment by badre429** [comment]
@varnastadeus  we have some issues when you use ng-select inside angular overlays system 
take look at this [https://stackblitz.com/edit/ng-select-5hunzp](https://stackblitz.com/edit/ng-select-5hunzp)
@angular/cdk is design agnostic framework for component common concerne functionality

**Comment by varnastadeus** [comment]
your content is overflowing and `overflow: hidden` is set. To solve that you need to have `overflow:visible` or use `appendTo="body"`.

**Comment by ArekSliwa** [comment]
I agree with @badre429  and @ghost . it should integrate with cdk overlay.  I would like to not create some 'hacks' for ng-select to achieve 'cdk overlay like' effect. For now I have dilemma: 
1. I would like to use ng-select because its great 
or
2. we use material/components and cdk and maybe we should drop ng-select because its outsider
